node/node.go

```diff
@@ -634,6 +634,73 @@ func (n *Node) OpenDatabaseWithFreezer(name string, cache, handles int, freezer,
 	return db, err
 }
 
+// The big difference is that the freezer inside was not running to avoid the mess between kvdb and freezer
+// while opening/closing db.
+// OpenDatabaseWithFreezerForPruneBlock opens an existing database with the given name (or
+// creates one if no previous can be found) from within the node's data directory,
+// also attaching a chain freezer to it. If the node is an ephemeral one, a memory database is returned.
+func (n *Node) OpenDatabaseWithFreezerForPruneBlock(name string, cache, handles int, freezer, namespace string, readonly bool) (ethdb.Database, error) {
```

Comment:
Please try not to invent new function, too many duplicated code.

Reply:
These new functions were the function without freeze go-routine running, to avoid the case that freeze go-routine did not stop while db closing.